Source/Renderer/FramebufferManager.js

```diff
@@ -0,0 +1,354 @@
+  index = defaultValue(index, 0);
+  //>>includeStart('debug', pragmas.debug);
+  if (index >= this._colorAttachmentsLength) {
+    throw new DeveloperError(
+      "index must be smaller than total number of color attachments."
+    );
+  }
+  //>>includeEnd('debug');
+  return this._colorTextures[index];
+};
+
+FramebufferManager.prototype.setColorTexture = function (texture, index) {
+  index = defaultValue(index, 0);
+  //>>includeStart('debug', pragmas.debug);
+  if (this._createColorAttachments) {
+    throw new DeveloperError(
+      "createColorAttachments must be false if setColorTexture is called."
+    );
+  }
+  if (index >= this._colorAttachmentsLength) {
+    throw new DeveloperError(
+      "index must be smaller than total number of color attachments."
+    );
+  }
+  //>>includeEnd('debug');
+  this._colorTextures[index] = texture;
+  this._attachmentsDirty = true;
+};
+
+FramebufferManager.prototype.getDepthRenderbuffer = function () {
+  return this._depthRenderbuffer;
+};
+
+FramebufferManager.prototype.setDepthRenderbuffer = function (renderbuffer) {
+  //>>includeStart('debug', pragmas.debug);
+  if (this._createDepthAttachments) {
+    throw new DeveloperError(
+      "createDepthAttachments must be false if setDepthRenderbuffer is called."
+    );
+  }
+  //>>includeEnd('debug');
+  this._depthRenderbuffer = renderbuffer;
+  this._attachmentsDirty = true;
+};
+
+FramebufferManager.prototype.getDepthTexture = function () {
+  return this._depthTexture;
+};
+
+FramebufferManager.prototype.setDepthTexture = function (texture) {
+  //>>includeStart('debug', pragmas.debug);
+  if (this._createDepthAttachments) {
+    throw new DeveloperError(
+      "createDepthAttachments must be false if setDepthTexture is called."
+    );
+  }
+  //>>includeEnd('debug');
+  this._depthTexture = texture;
+  this._attachmentsDirty = true;
+};
+
+FramebufferManager.prototype.getDepthStencilRenderbuffer = function () {
+  return this._depthStencilRenderbuffer;
+};
+
+FramebufferManager.prototype.setDepthStencilRenderbuffer = function (
+  renderbuffer
+) {
+  //>>includeStart('debug', pragmas.debug);
+  if (this._createDepthAttachments) {
+    throw new DeveloperError(
+      "createDepthAttachments must be false if setDepthStencilRenderbuffer is called."
+    );
+  }
+  //>>includeEnd('debug');
+  this._depthStencilRenderbuffer = renderbuffer;
+  this._attachmentsDirty = true;
+};
+
+FramebufferManager.prototype.getDepthStencilTexture = function () {
+  return this._depthStencilTexture;
+};
+
+FramebufferManager.prototype.setDepthStencilTexture = function (texture) {
+  //>>includeStart('debug', pragmas.debug);
+  if (this._createDepthAttachments) {
+    throw new DeveloperError(
+      "createDepthAttachments must be false if setDepthStencilTexture is called."
+    );
+  }
+  //>>includeEnd('debug');
+  this._depthStencilTexture = texture;
+  this._attachmentsDirty = true;
+};
+
+FramebufferManager.prototype.clear = function (
```

Comment:
@lilleyse I thought it didn't seem right to have `clear` take in a clear color argument, and from the comments above always temporarily use the clear command's framebuffer during the `execute` call.

Reply:
Good with me